HttpSignature.md

```diff
@@ -1,40 +1,160 @@
 
 ```text
 Client                          KeyID                            Resource
 App                          Document                            Server
 |                                |                                   |  
-|-request URL ------------------------------------------------------>| 
-|<-------------------------------------- 401 + WWW-Auth Sig header---|
+|-(1) request URL -------------------------------------------------->| 
+|<============(2) 40x + WWW-Authenticate: HttpSig + (Link) to ACL ===|
+|                                |                                   |
+| (choose key)                   |                                   |
 |                                |                                   |
-|-- sign headers+keyId---------------------------------------------->|
+|-(3)- sign headers+keyId------------------------------------------->|
 |                                |                       initial auth|
 |                                |                       verification| 
 |                                |                                   |
-|                                |<----------------------GET keyId---| 
-|                                |-return keyId doc----------------->|
-|                                                        -verify sig |
-|                                                        -verify wACL|
+|                                |<-----------------(4) GET keyId----| 
+|                                |-(5) return keyId doc------------->|
+|                                                       - verify sig |
+|                                                       - verify ACL |
 |                                                                    |
-|<-------------------------------------------------answer resource---|
+|<---------------------------------------------(6) answer resource---|
 ```                                                                   
 
-The addition to the HTTP signature protocol is the request by the
-resource server for the `keyId document`. 
+In (2) the Resource server responds to a request by sending a 401 or 402 response with the `WWW-Authenticate: HttpSig` header. The `WWW-Authenticate` header is specified in §[4.1 of HTTP/1.1](https://www.rfc-editor.org/rfc/rfc7235#section-4.1).  The `HttpSig` challenge method needs to be defined here (todo) and registered as specified in the [Authentication Scheme Registry](https://www.rfc-editor.org/rfc/rfc7235#section-5.1).
+A message from a server can look like this:
+
+```HTTP
+HTTP/1.1 401 Unauthorized
+WWW-Authenticate: HttpSig
+Link: </comments/.acl>; rel="acl"
+```
+
+It should also add a `Link:` relation to an Access-Control resource which describes which resources might be accessible. 
+This is described in [Web Access Control Spec](https://github.com/solid/web-access-control-spec/).
+(Without such a Link the client would only be able to guess what key to send.)
```

Comment:
The `WWW-Authenticate: HttpSig` header is sent by the server to the client.  The `HttpSig` method is meant to indicate to the client that it can use "Signing Http Messages" with the `keyId` used as URL.
It could send the content of the ACL in the body so not requiring an extra connection (or with HTTP/2 the server can send it ahead of being asked)